mdagent/tools/base_tools/analysis_tools/distance_tools.py

```diff
@@ -0,0 +1,565 @@
+    TRAJECTORY_FILEID_DESC,
+)
+
+
+class distanceToolsUtils:
+    def __init__(self, path_registry: Optional[PathRegistry] = None):
+        self.path_registry = path_registry
+
+    def all_possible_pairs(self, array1, array2):
+        return list(itertools.product(array1, array2))
+
+    def calc_residue_dist(self, residues=(0, 0)):
+        """
+        Return the C-alpha distance between two residues.
+        returns: float  distance
+        """
+        diff_vector = residues[0] - residues[1]
+        return np.sqrt(np.vdot(diff_vector, diff_vector))
+
+    def calc_side_center_mass(self, traj):
+        """
+        Return approximate center of mass of each side chain per frame.
+        COM od Glycine is approximated by the coordinate of its CA atom.
+
+        Note: Because compute_center_of_mass gets the center of mass of only one
+        selection at a time, we need to loop over all residues to get the each COM.
+
+        returns: com_matrix: np.array, shape=(n_frames, n_residues, 3)
+        """
+        traj = traj.atom_slice(traj.top.select("protein"), inplace=False)
+        com_matrix = np.zeros((traj.n_frames, traj.n_residues, 3))
+        for i in range(0, traj.topology.n_residues):
+            selection = f"(resid {i}) and sidechain"
+            if len(selection) < 1:
+                selection = f"(resid {i}) and (name CA)"
+            try:
+                com_matrix[:, i, :] = md.compute_center_of_mass(traj, select=selection)
+            except Exception as e:
+                raise (f"Error calculating center of mass for residue, {i}: {str(e)}")
+        return com_matrix
+
+    def calc_dis_matrix_from_com_all_resids(self, traj, com_matrix):
+        """
+        Return the distance matrix between the center of mass of residues
+
+        returns: dis_matrix: np.array, shape=(n_frames, n_residues, n_residues)
+        """
+        new_xyz = com_matrix
+        new_topology = pd.DataFrame(
+            {
+                "serial": range(traj.n_residues),
+                "name": ["COM" for _ in traj.topology.residues],
+                "resSeq": range(traj.n_residues),
+                "resName": [i.name for i in traj.topology.residues],
+                "element": ["VS" for _ in traj.topology.residues],
+                "chainID": [
+                    i.chain.chain_id if i.chain.chain_id else 0
+                    for i in traj.topology.residues
+                ],
+                "segmentID": [i.segment_id for i in traj.topology.residues],
+            }
+        )
+        top = md.Topology.from_dataframe(new_topology)
+        new_traj = md.Trajectory(new_xyz, top)
+        pairs = self.all_possible_pairs(range(traj.n_residues), range(traj.n_residues))
+        _dis_matrix = md.compute_distances(new_traj, pairs)
+        dis_matrix = md.geometry.squareform(_dis_matrix, residue_pairs=pairs)
+        return dis_matrix
+
+    def calc_residue_side_dist(self, traj, frame, residue_one, residue_two):
```

Comment:
mmm. This is from one of my first local versions of the tool, where i was deciding how to make this work. You're right, right now is not been used. I could drop them, or keep them in case we would want to add or change a tool later on?